### DISCUSSION TITLE
[Question] What does the default offset 0.5 mean in official observation function 'height_scan()' ?

### DISCUSSION THREAD

**Discussion by AlexanderAbernathy** [isaac-sim/IsaacLab#774] [Q&A] [answered]
### Question

I was trying to get the terrain height of the rough terrain for RL these days. I found a sensor with an official function called 'height_scan()' in observation, and I found a 0.5 offset by default in the function. Users can directly modify it. The codes are shown below.

```python
def height_scan(env: ManagerBasedEnv, sensor_cfg: SceneEntityCfg, offset: float = 0.5) -> torch.Tensor:
    """Height scan from the given sensor w.r.t. the sensor's frame.

    The provided offset (Defaults to 0.5) is subtracted from the returned values.
    """
    # extract the used quantities (to enable type-hinting)
    sensor: RayCaster = env.scene.sensors[sensor_cfg.name]
    # height scan: height = sensor_height - hit_point_z - offset
    return sensor.data.pos_w[:, 2].unsqueeze(1) - sensor.data.ray_hits_w[..., 2] - offset
```
What does this offset mean? Should I change this value when using different robots?

**Comment by Mayankm96** [comment, marked as answer]
Good question. We took this hyperparameter from the [legged_gym](https://github.com/leggedrobotics/legged_gym/blob/master/legged_gym/envs/base/legged_robot.py#L222) implementation. I suppose the initial motivation was to keep the plane ground at "0" observation value so an offset was added to account for that.

Judging that it works for different robots means the policy learns to deal with this bias, and it may not really be necessary.

**Reply by AlexanderAbernathy** [reply]
Thanks very much for your answers.
However, I am still a little confused with some data. I've noticed that in 'ObservationsCfg', the observation items are set with noise. Does this mean these settings are globally effective even if we get these values anywhere in the project? (for example, in the functions of reward.py, we can get some robot data by using 'asset : RigidObject | Articulation = env.scene[asset_cfg.name]', it includes the body linear velocities, joint velocities, or other data values like in observations.) Are these data with noise and consistent with the observations?.
In my own reward function, I copied the codes from 'height_scan()' but without using the offset.  According to the data that I derived, I thought that there was not any noise at all. If I wanna use the data directly from the observations to ensure data consistency for my own reward functions and keep the data, what should I do?
